### DISCUSSION TITLE
Use a string translation tool ✏

### DISCUSSION THREAD

**Discussion by giant-teapot** [fvtt-fria-ligan/forbidden-lands-foundry-vtt#32] [Ideas]
There are tools out there that could help us know the state of the system's translation (% of missing strings) as well as offer a interface to ease external contributions.

I've been toying with [GitLocalize](https://gitlocalize.com) on my fork and it's not too bad. In a few words:

- it's free for us
- you login with your github account
- it took me literally 3 minutes to set up
- you can use the website to translate strings and then you can create a pull request with those translations
- you can see one such PR here: https://github.com/giant-teapot/forbidden-lands-foundry-vtt/pull/1

A few screenshots to give you an idea:

![translation_panel](https://user-images.githubusercontent.com/477392/107874351-458ff700-6eb9-11eb-931b-8cfb419e5c24.PNG)

![translation_tool](https://user-images.githubusercontent.com/477392/107874352-46c12400-6eb9-11eb-800f-a782a922f408.PNG)

I think it would be a good thing for the project:

- With a bit of documentation, I hope it would attract more contributors (and of more diverse profiles).
- In turn, those contributors would help us maintain the growing number of strings, and perhaps extend support to other languages…
- Also, it helps to see which languages are lagging behind.

Given the effort it requires to set up, it seems to me like a juicy low-hanging fruit. 
What do you think?

**Comment by jimorie** [comment]
Looks nice at first glance! :+1:

**Comment by aMediocreDad** [comment]
I'll give this a whirl!

EDIT 1: Hmm. Seems to be stuck on submitting. Created a new branch and made sure paths are correct. But it fails to proceed past submit. Will try a different browser.

EDIT 2: It was a browser issue.

**Reply by giant-teapot** [reply]
Just for kicks, I sent out PR #33.
A few quirks:

- it does not like empty lines
- it always adds new keys at the end of the file

**Reply by aMediocreDad** [reply]
If this makes localization more effortless I am fine with those trade-offs.
